ibis/backends/pandas/executor.py

```diff
@@ -110,19 +88,71 @@ def visit(cls, op: ops.RandomScalar):
 
     @classmethod
     def visit(cls, op: ops.Greatest, arg):
-        return columnwise(lambda df: df.max(axis=1), arg)
+        return cls.columnwise(lambda df: df.max(axis=1), arg)
 
     @classmethod
     def visit(cls, op: ops.Least, arg):
-        return columnwise(lambda df: df.min(axis=1), arg)
+        return cls.columnwise(lambda df: df.min(axis=1), arg)
 
     @classmethod
     def visit(cls, op: ops.Coalesce, arg):
-        return columnwise(lambda df: df.bfill(axis=1).iloc[:, 0], arg)
+        return cls.columnwise(lambda df: df.bfill(axis=1).iloc[:, 0], arg)
 
     @classmethod
     def visit(cls, op: ops.Value, **operands):
-        return pick_kernel(op, operands)
+        # automatically pick the correct kernel based on the operand types
+        typ = type(op)
+        name = op.name
+        dtype = PandasType.from_ibis(op.dtype)
+        kwargs = {"operands": operands, "name": name, "dtype": dtype}
+
+        # decimal operations have special implementations
+        if op.dtype.is_decimal():
+            func = cls.kernels.elementwise_decimal[typ]
+            return cls.elementwise(func, **kwargs)
+
+        # prefer generic implementations if available
+        if func := cls.kernels.generic.get(typ):
+            return cls.generic(func, **kwargs)
+
+        first, *rest = operands.values()
```

Comment:
Looks like `first` might not be used.

```suggestion
        _, *rest = operands.values()
```

Reply:
Updated.